src/promise.rs

```diff
@@ -342,7 +429,15 @@ impl<T: Send + 'static> PromiseImpl<T> {
         }
     }
 
-    fn block_until_ready_mut(&mut self) -> &mut T {
+    #[allow(unused_variables)]
+    fn block_until_ready_mut(&mut self, task_type: TaskType) -> &mut T {
+        // Constantly poll until we're ready.
+        #[cfg(feature = "smol")]
+        while self.poll(task_type).is_pending() {
+            // Tick unless poll does it for us.
+            #[cfg(not(feature = "smol_tick_poll"))]
+            Self::tick(task_type);
+        }
```

Comment:
Generally a bad idea to tick the executor in a loop like this. You may want to consider using the [`block_on`](https://docs.rs/futures-lite/latest/futures_lite/future/fn.block_on.html) function with the [`tick`](https://docs.rs/async-executor/latest/async_executor/struct.Executor.html#method.tick) future to avoid a spinloop.